src/ifcparse/IfcSpfHeader.cpp

```diff
@@ -36,19 +36,22 @@ static const char * const DATA				= "DATA";
 using namespace IfcParse;
 
 void IfcSpfHeader::readSemicolon() {
-	if (!TokenFunc::isOperator(_lexer->Next(), ';')) {
+    Token next = _lexer->Next();
+    if (next.lexer != 0 && !TokenFunc::isOperator(next, ';')) {
```

Comment:
Thanks for this, I had a look as well. This only protects the header parser. I think that in the data section this issue less likely to manifest itself, as there eof is detected and non complete instances are not registered in the map. Still, I think it might be better to resolve it in `asStringRef()` to reduce the likelihood of hard segfaults, by simply throwing an exception there it seems the issue is addressed as well (also, the check on lexer is a bit cryptic, better check on `type == NONE`).

I suggest the following instead at the start of `asStringRef()`:

~~~C++
	if (t.type == Token_NONE) {
		throw IfcParse::IfcException("Null token encountered, premature end of file?");
	}
~~~